src/libraries/Microsoft.PowerFx.Core/Entities/External/TableParameters.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    public class TableParameters
+    {
+        public SortRestrictions SortRestriction { get; init; }
+
+        public FilterRestrictions FilterRestriction { get; init; }
+
+        public SelectionRestrictions SelectionRestriction { get; init; }
+
+        public GroupRestrictions GroupRestriction { get; init; }
+
+        public IEnumerable<string> FilterFunctions { get; init; }
```

Comment:
>string [](http://example.com/codeflow?start=27&length=6)

scrutinize this - can we comment these?

we can always keep something internal (if it's not needed) and make public later.

Reply:
Will be uncertain as none of this is documented in PA

Reply:
If we don't know what it is, then keep it internal.

Reply:
I have improved documentation